RecoParticleFlow/PFClusterProducer/src/Cluster3DPCACalculator.cc

```diff
@@ -0,0 +1,132 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/isFinite.h"
+
+#include <cmath>
+#include <unordered_map>
+
+#include "vdt/vdtMath.h"
+
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHit.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+
+#include "TPrincipal.h"
+
+class Cluster3DPCACalculator : public PFCPositionCalculatorBase {
+ public:
+  Cluster3DPCACalculator(const edm::ParameterSet& conf) :
+    PFCPositionCalculatorBase(conf),    
+    _pca(new TPrincipal(3,"D")){ 
+  }
+  Cluster3DPCACalculator(const Cluster3DPCACalculator&) = delete;
+  Cluster3DPCACalculator& operator=(const Cluster3DPCACalculator&) = delete;
+
+  void calculateAndSetPosition(reco::PFCluster&);
+  void calculateAndSetPositions(reco::PFClusterCollection&);
+
+ private:
+  std::unique_ptr<TPrincipal> _pca;
```

Comment:
This variable does not follow CMS coding conventions which only allow underscore at the end. Variables beginning with an underscore are often reserved for the compiler which is why we avoid them.